zbus/src/match_rule.rs

```diff
@@ -144,15 +145,16 @@ impl<'m> MatchRule<'m> {
     }
 
     /// Match messages whose first argument is within the specified namespace.
-    ///
-    /// Note that while the spec allows this to be any string that's a valid bus or interface name
-    /// except that it can have no `.`, we only allow valid interface names. The reason is not only
-    /// to keep things simple and type safe at the same time but also for the fact that use cases of
-    /// only matching on the first component of a bus or interface name are unheard of.
```

Comment:
* These docs need to be moved to (maybe modified to fit) the new methods.
* Let's document here why this is deprecated.

Reply:
> These docs need to be moved to (maybe modified to fit) the new methods.

I don't see how this comment applies to the new methods...

Reply:
> I don't see how this comment applies to the new methods...

Not as is, no (hence the "maybe modified to fit" bit). :) Anyway, I think this is covered by the other thread.

Would have been great to address the second bullet here but it's not biggie I guess.

Reply:
> Would have been great to address the second bullet here but it's not biggie I guess.

But I think I did address it :)